scripts/process/geom_dataframe.R

```diff
@@ -0,0 +1,68 @@
+#' from a sp (will be sf) object, optionally clip and optionally 
+#' mutate data into svg element attributes
+#' 
+#' @details 
+#' This function takes a single required dependency, which is `spatial_data`
+#' Presently, `spatial_data` is an `sf` object.
+#' if `attributes` are present, they are used to add to or modify the `data.frame`, and 
+#' they are the only variables included in the return data (other than `geometry`).
+#' in `spatial_data`. Supported operations are specifying a string to be repeated, or 
+#' a string substitution using information from another column with the "{{variable}}" 
+#' (`mustache`) syntax. This operation is skipped if `attributes` isn't present.
+#' if `clip_box` is present as a dependency, it is used to clip the geometries in 
+#' `spatial_data`
+#' See `sub_attribute_texte` function.
+process.geom_dataframe <- function(viz){
+  deps <- readDepends(viz)
+  checkRequired(deps, 'spatial_data') # spatial data is the only required depends
+  
+  geom_dataframe <- deps[['spatial_data']]
+  
+  if (!is.null(deps[['clip_box']])){
+    # remove all polygons that are outside the box
+    # trim all polgons that overlap the box
+    geom_dataframe <- st_intersection(st_buffer(geom_dataframe, 0), deps[['clip_box']])
```

Comment:
hah. Just going for the 0 buffer. 👍 For whatever reason maps::map("state") causes the issue here.